<?xml version="1.0"?>
<launch>
    
    <arg name="trans_list" default="None" />
    <arg name="path" default="/home/mikulas/crazyflie_ws/src/crazyflie_ros/crazyflie_demo/src/" />
    <arg name="n_swarm" default="10" /> 
    
    <group ns="crazyflie1">
        <node pkg="crazyflie_demo" type="execute_swarm_transitions.py" name="full_state_swarm" output="screen">
            <param name="n_agent" value="0" />
            <param name="pub_topic" value="cmd_full_state" />
            <param name="frame_id" value="/world" />
            <param name="path" value="$(arg path)" />
            <param name="transitions_list" value="$(arg trans_list)"/>
        </node>
        
        <node pkg="crazyflie_demo" type="marker_rviz_swarm.py" name="marker_republisher" output="screen">
            <param name="n_agent" value="0" />
            <param name="sub_topic" value="cmd_full_state" />
            <param name="frame_id" value="/world" />
            <param name="n_swarm" value="$(arg n_swarm)" />
        </node>
    </group>
      
    <group ns="crazyflie2">
        <node pkg="crazyflie_demo" type="execute_swarm_transitions.py" name="full_state_swarm"  output="screen">
            <param name="n_agent" value="1" />
            <param name="pub_topic" value="cmd_full_state" />
            <param name="frame_id" value="/world" />
            <param name="path" value="$(arg path)" />
            <param name="transitions_list" value="$(arg trans_list)"/>>
        </node>
        
        <node pkg="crazyflie_demo" type="marker_rviz_swarm.py" name="marker_republisher" output="screen">
            <param name="n_agent" value="1" />
            <param name="sub_topic" value="cmd_full_state" />
            <param name="frame_id" value="/world" />
            <param name="n_swarm" value="$(arg n_swarm)" />
        </node>
    </group>
    
    <group ns="crazyflie3">
        <node pkg="crazyflie_demo" type="execute_swarm_transitions.py" name="full_state_swarm"  output="screen">
            <param name="n_agent" value="2" />
            <param name="pub_topic" value="cmd_full_state" />
            <param name="frame_id" value="/world" />
            <param name="path" value="$(arg path)" />
            <param name="transitions_list" value="$(arg trans_list)"/>
        </node>
        
        <node pkg="crazyflie_demo" type="marker_rviz_swarm.py" name="marker_republisher" output="screen">
            <param name="n_agent" value="2" />
            <param name="sub_topic" value="cmd_full_state" />
            <param name="frame_id" value="/world" />
            <param name="n_swarm" value="$(arg n_swarm)" />
        </node>
    </group>
    
    <group ns="crazyflie4">
        <node pkg="crazyflie_demo" type="execute_swarm_transitions.py" name="full_state_swarm"  output="screen">
            <param name="n_agent" value="3" />
            <param name="pub_topic" value="cmd_full_state" />
            <param name="frame_id" value="/world" />
            <param name="path" value="$(arg path)" />
            <param name="transitions_list" value="$(arg trans_list)"/>
        </node>
        
        <node pkg="crazyflie_demo" type="marker_rviz_swarm.py" name="marker_republisher" output="screen">
            <param name="n_agent" value="3" />
            <param name="sub_topic" value="cmd_full_state" />
            <param name="frame_id" value="/world" />
            <param name="n_swarm" value="$(arg n_swarm)" />
        </node>
    </group>
    
    <group ns="crazyflie5">
        <node pkg="crazyflie_demo" type="execute_swarm_transitions.py" name="full_state_swarm"  output="screen">
            <param name="n_agent" value="4" />
            <param name="pub_topic" value="cmd_full_state" />
            <param name="frame_id" value="/world" />
            <param name="path" value="$(arg path)" />
            <param name="transitions_list" value="$(arg trans_list)"/>
        </node>
        
        <node pkg="crazyflie_demo" type="marker_rviz_swarm.py" name="marker_republisher" output="screen">
            <param name="n_agent" value="4" />
            <param name="sub_topic" value="cmd_full_state" />
            <param name="frame_id" value="/world" />
            <param name="n_swarm" value="$(arg n_swarm)" />
        </node>
    </group>
    
    <group ns="crazyflie6">
        <node pkg="crazyflie_demo" type="execute_swarm_transitions.py" name="full_state_swarm"  output="screen">
            <param name="n_agent" value="5" />
            <param name="pub_topic" value="cmd_full_state" />
            <param name="frame_id" value="/world" />
            <param name="path" value="$(arg path)" />
            <param name="transitions_list" value="$(arg trans_list)"/>
        </node>
        
        <node pkg="crazyflie_demo" type="marker_rviz_swarm.py" name="marker_republisher" output="screen">
            <param name="n_agent" value="5" />
            <param name="sub_topic" value="cmd_full_state" />
            <param name="frame_id" value="/world" />
            <param name="n_swarm" value="$(arg n_swarm)" />
        </node>
    </group>
    
    <group ns="crazyflie7">
        <node pkg="crazyflie_demo" type="execute_swarm_transitions.py" name="full_state_swarm"  output="screen">
            <param name="n_agent" value="6" />
            <param name="pub_topic" value="cmd_full_state" />
            <param name="frame_id" value="/world" />
            <param name="path" value="$(arg path)" />
            <param name="transitions_list" value="$(arg trans_list)"/>
        </node>
        
        <node pkg="crazyflie_demo" type="marker_rviz_swarm.py" name="marker_republisher" output="screen">
            <param name="n_agent" value="6" />
            <param name="sub_topic" value="cmd_full_state" />
            <param name="frame_id" value="/world" />
            <param name="n_swarm" value="$(arg n_swarm)" />
        </node>
    </group>
    
    <group ns="crazyflie8">
        <node pkg="crazyflie_demo" type="execute_swarm_transitions.py" name="full_state_swarm"  output="screen">
            <param name="n_agent" value="7" />
            <param name="pub_topic" value="cmd_full_state" />
            <param name="frame_id" value="/world" />
            <param name="path" value="$(arg path)" />
            <param name="transitions_list" value="$(arg trans_list)"/>
        </node>
        
        <node pkg="crazyflie_demo" type="marker_rviz_swarm.py" name="marker_republisher" output="screen">
            <param name="n_agent" value="7" />
            <param name="sub_topic" value="cmd_full_state" />
            <param name="frame_id" value="/world" />
            <param name="n_swarm" value="$(arg n_swarm)" />
        </node>
    </group>
    
    <group ns="crazyflie9">
        <node pkg="crazyflie_demo" type="execute_swarm_transitions.py" name="full_state_swarm"  output="screen">
            <param name="n_agent" value="8" />
            <param name="pub_topic" value="cmd_full_state" />
            <param name="frame_id" value="/world" />
            <param name="path" value="$(arg path)" />
            <param name="transitions_list" value="$(arg trans_list)"/>
        </node>
        
        <node pkg="crazyflie_demo" type="marker_rviz_swarm.py" name="marker_republisher" output="screen">
            <param name="n_agent" value="8" />
            <param name="sub_topic" value="cmd_full_state" />
            <param name="frame_id" value="/world" />
            <param name="n_swarm" value="$(arg n_swarm)" />
        </node>
    </group>
    
    <group ns="crazyflie10">
        <node pkg="crazyflie_demo" type="execute_swarm_transitions.py" name="full_state_swarm"  output="screen">
            <param name="n_agent" value="9" />
            <param name="pub_topic" value="cmd_full_state" />
            <param name="frame_id" value="/world" />
            <param name="path" value="$(arg path)" />
            <param name="transitions_list" value="$(arg trans_list)"/>
        </node>
        
        <node pkg="crazyflie_demo" type="marker_rviz_swarm.py" name="marker_republisher" output="screen">
            <param name="n_agent" value="9" />
            <param name="sub_topic" value="cmd_full_state" />
            <param name="frame_id" value="/world" />
            <param name="n_swarm" value="$(arg n_swarm)" />
        </node>
    </group>
    
    <!-- run rviz
    <node name="rviz" pkg="rviz" type="rviz" args="-d $(find crazyflie_demo)/launch/crazyflie_pos_multi.rviz" required="true" />
    -->
    
 </launch>
